public/app/features/alerting/unified/components/receivers/TemplateForm.tsx

```diff
@@ -131,36 +140,31 @@ export const TemplateForm = ({ existing, alertManagerSourceName, config, provena
   } = formApi;
 
   const submit = async (values: TemplateFormValues) => {
-    const returnLink = makeAMLink('/alerting/notifications', alertManagerSourceName, {
+    const returnLink = makeAMLink('/alerting/notifications', alertmanager, {
       tab: ContactPointsActiveTabs.NotificationTemplates,
     });
 
     try {
-      if (!existing) {
-        await createNewTemplate({ template: values });
+      if (!originalTemplate) {
+        await createNewTemplate({ templateValues: values });
       } else {
-        await updateTemplate({ originalName: existing.name, template: values });
+        await updateTemplate({ template: originalTemplate, patch: values });
       }
+      appNotification.success('Template saved', `Template ${values.title} has been saved`);
       locationService.push(returnLink);
     } catch (error) {
       appNotification.error('Error saving template', stringifyErrorLike(error));
```

Comment:
We end up showing the error twice in this case when using the k8s API:

<img width="640" alt="image" src="https://github.com/user-attachments/assets/c2efea83-bde3-4812-b249-50546719584a">

Shall we suppress all messages from the hook at the k8s API level? And then components can handle it themselves (I think @gillesdemey was thinking this approach as well)